lib/puppetserver/ca/action/prune.rb

```diff
@@ -50,15 +50,22 @@ def run(inputs)
           loader = X509Loader.new(puppet.settings[:cacert], puppet.settings[:cakey], puppet.settings[:cacrl])
 
           puppet_crl = loader.crls.select { |crl| crl.verify(loader.key) }
-          prune_CRLs(puppet_crl)
-          update_pruned_CRL(puppet_crl, loader.key)
-          FileSystem.write_file(puppet.settings[:cacrl], loader.crls, 0644)
+          update_flag = prune_CRLs(puppet_crl)
+
+          if update_flag == true
+            update_pruned_CRL(puppet_crl, loader.key)
+            FileSystem.write_file(puppet.settings[:cacrl], loader.crls, 0640)
```

Comment:
Is there a reason you changed the file permissions here? I'm not sure off the top of my head whether 'other' needs read permissions.. If not, then this change is good.

Reply:
Ah, I was testing out why the error out parsing was happening and did some test with how the file was written out. I forgot to change it back to 0644.